c10/cuda/CUDAStream.cpp

```diff
@@ -175,12 +195,12 @@ static void initGlobalStreamState() {
   num_gpus = device_count();
```

Comment:
Actually after seeing this I am pretty confident that we should use a compile time max GPU count to avoid the std::unique_ptr<x[]> complexity and use static arrays directly. Realistically one is not going to be using more than 16 GPUs and it saves a lot of potential pitfalls in the general C++ programming sense.